### ISSUE TITLE
Error while importing GCode

### ISSUE THREAD

**Issue by callimero** [Heinz-Loepmeier/nozzleboss#2]
What can be the problem?

![image](https://user-images.githubusercontent.com/7713008/163338002-31ec1144-c21c-4832-b942-c088d05f927f.png)

Blender 3.0.0

**Comment by Heinz-Loepmeier** [comment]
@callimero
Can you upload the G-code somewhere? 
Easier to test, I think i might know what the problem is here.

**Comment by callimero** [comment]
I am on vacation now. I will look into it when back.

**Comment by rustyawhile001** [comment]
I am getting similar error with importing the following:
;nozzleboss
G28 ;homing
M104 S180 ;set hotend temp
M190 S50 ;wait for bed temp
M109 S200 ;wait for hotendtemp
M83; relative extrusion mode (REQUIRED)


G10 
G1 F3600
G1 Y10.0 
G1 F1800
G11 
T0; switch to extruder T0 (any G-code macro can be passed here)

G1 X-10.0 Y0.0 E3.5278 F1799
G1 X0.0 Y-10.0 E3.5278 F1799
G1 X10.0 Y0.0 E3.5278 F1799
G1 X0.0 Y10.0 E3.5278 F1799

G10 ;retract
M104 S0 ;deactivate hotend
M140 S0 ;deactivate bed
G28 ;homing
M84 ;turn off motors

![import_error](https://github.com/user-attachments/assets/43b3a5d2-a34b-4082-aa9e-ff9f3f9a6594)

I noticed that if the initial curve vertex is above the object origin (say Z =0.3) then the import works well.. but when Z = 0.0 (same as the object origin), then no thickness is render on the import. It seems like the g-code needs a G1 Z declration initially.. the following import works:
;nozzleboss
G28 ;homing
M104 S180 ;set hotend temp
M190 S50 ;wait for bed temp
M109 S200 ;wait for hotendtemp
M83; relative extrusion mode (REQUIRED)


G10 
G1 F3600
G1 Y10.0 Z1.0
G1 F1800
G11 
T0; switch to extruder T0 (any G-code macro can be passed here)

G1 X-10.0 Y0.0 E3.5278 F1799
G10 
G1 F3600
G1 X0.0 Y20.0 
G1 F1800
G11 
G1 X-10.0 Y10.0 E3.5278 F1799

G10 ;retract
M104 S0 ;deactivate hotend
M140 S0 ;deactivate bed
G28 ;homing
M84 ;turn off motors

**Comment by Heinz-Loepmeier** [comment]
Yes without any Z-coord there is no way of auto detecting any layerheight.
It gives an error, but should still import the 2D path without faces. (Note flow and speedmaps are face corner attribute maps that don't work on meshes without faces)
With G1 Z1 your layer height can be calculated as distance to origin.

If you don't want to deal with layer height info in the mesh you can use https://github.com/Heinz-Loepmeier/Blender-Gcode-Import (if you use it in combination with nozzleboss, pls download nozzleboss dev version https://github.com/Heinz-Loepmeier/nozzleboss-dev ,otherwise you get a name clash between nozzleboss and gcode importer addon)

The simple G-code importer works well with Alessandro Zomperelli's https://github.com/alessandro-zomparelli/gcode-exporter.
